packages/core/src/messages/responses/vector/vector-get-item-metadata-batch.ts

```diff
@@ -0,0 +1,84 @@
+import {SdkError} from '../../../errors';
+import {ResponseBase, ResponseError, ResponseSuccess} from '../response-base';
+
+/**
+ * Parent response type for a VectorGetItemMetadataBatch request. The
+ * response object is resolved to a type-safe object of one of
+ * the following subtypes:
+ *
+ * - {Success}
+ * - {Error}
+ *
+ * `instanceof` type guards can be used to operate on the appropriate subtype.
+ * @example
+ * For example:
+ * ```
+ * if (response instanceof VectorGetItemMetadataBatch.Error) {
+ *   // Handle error as appropriate.  The compiler will smart-cast `response` to type
+ *   // `VectorGetItemMetadataBatch.Error` in this block, so you will have access to the properties
+ *   // of the Error class; e.g. `response.errorCode()`.
+ * }
+ * ```
+ */
+export abstract class Response extends ResponseBase {
+  hits():
```

Comment:
just curious about the choice of the word `hit` for these. Is that a term of art in this space? I'd probably have just expected `values` or `items`.

Reply:
In search, `hits` are the results of a search operation. While this is more of a lookup than a search, I was thinking about that and how we have "cache hits" and "cache misses"; each item is either a hit or a miss in the index. Thus an object with only the hits I called `hits`. That said I was gnashing my teeth over what to call it.

Reply:
ah, so I guess these APIs are not guaranteed to return the same number of results as the number of input IDs? that makes some sense. I feel like I still would have expected `items` or `rows` or `values`, before `hits` but I don't have a huge problem with `hits`, just offering up my unfiltered reaction :)

Reply:
> ah, so I guess these APIs are not guaranteed to return the same number of results as the number of input IDs?

That's right. The methods are akin to the dictionary get fields operations in that way.

> I feel like I still would have expected items or rows or values, before hits but I don't have a huge problem with hits, just offering up my unfiltered reaction :)

<img src="https://media1.giphy.com/media/ZPtSHbjmS8x7G/giphy.gif"/>